src/adaptation/stepsize.jl

```diff
@@ -2,31 +2,43 @@
 ### Mutable states ###
 ######################
 
-mutable struct DAState{T<:AbstractFloat}
+mutable struct DAState{T<:AbstractScalarOrVec{<:AbstractFloat}}
     m     :: Int
     ϵ     :: T
     μ     :: T
     x_bar :: T
     H_bar :: T
 end
 
+computeμ(ϵ::AbstractScalarOrVec{<:AbstractFloat}) = log.(10 * ϵ)
+
 function DAState(ϵ::AbstractFloat)
     μ = computeμ(ϵ)
     return DAState(0, ϵ, μ, 0.0, 0.0)
 end
 
-function computeμ(ϵ::AbstractFloat)
-    return log(10 * ϵ) # see NUTS paper sec 3.2.1
+function DAState(ϵ::AbstractVector{<:AbstractFloat})
+    n = length(ϵ)
+    μ = computeμ(ϵ)
+    return DAState(0, ϵ, μ, zeros(n), zeros(n))
 end
 
 function reset!(dastate::DAState{T}) where {T<:AbstractFloat}
-    dastate.μ = computeμ(dastate.ϵ)
     dastate.m = 0
+    dastate.μ = computeμ(dastate.ϵ)
     dastate.x_bar = zero(T)
     dastate.H_bar = zero(T)
 end
 
-mutable struct MSSState{T<:AbstractFloat}
+function reset!(dastate::DAState{VT}) where {VT<:AbstractVector{<:AbstractFloat}}
```

Comment:
As this doesn't hurt performance, I tend to make struct have a type `T` only if we need to dispatch through the type; otherwise get it using `eltype`.

Reply:
Oh I know what you mean. Good idea.